src/canopy_utils_mod.F90

```diff
@@ -125,4 +125,130 @@ function CalcFlameH(frp, dx)
+        ! !INTERFACE:
+
+        ! !INPUT PARAMETERS:
+        integer,  INTENT(IN) :: co2_opt       ! Option for co2 inhibition calculation
+        ! 0=Possell & Hewitt (2011);
+        ! 1=Wilkinson et al. (2009)
+        ! >1=off
+        real(rk), INTENT(IN) :: co2_set       ! User set atmospheric CO2 conc [ppmv]
+        !
+        ! !RETURN VALUE:
+!        REAL(rk)             :: GAMMA_CO2  ! CO2 activity factor [unitless]
+        !
+        ! !LOCAL VARIABLES:
+        REAL(rk)             :: CO2i       ! Intercellular CO2 conc [ppmv]
+        REAL(rk)             :: ISMAXi     ! Asymptote for intercellular CO2
+        REAL(rk)             :: HEXPi      ! Exponent for intercellular CO2
+        REAL(rk)             :: CSTARi     ! Scaling coef for intercellular CO2
+        REAL(rk)             :: ISMAXa     ! Asymptote for atmospheric CO2
+        REAL(rk)             :: HEXPa      ! Exponent for atmospheric CO2
+        REAL(rk)             :: CSTARa     ! Scaling coef for atmospheric CO2
+        !
+        ! !REMARKS:
+        !  References:
+        !  ============================================================================
+        !  (1 ) Heald, C. L., Wilkinson, M. J., Monson, R. K., Alo, C. A.,
+        !       Wang, G. L., and Guenther, A.: Response of isoprene emission
+        !       to ambient co(2) changes and implications for global budgets,
+        !       Global Change Biology, 15, 1127-1140, 2009.
+        !  (2 ) Wilkinson, M. J., Monson, R. K., Trahan, N., Lee, S., Brown, E.,
+        !       Jackson, R. B., Polley, H. W., Fay, P. A., and Fall, R.: Leaf
+        !       isoprene emission rate as a function of atmospheric CO2
+        !       concentration, Global Change Biology, 15, 1189-1200, 2009.
+        !  (3 ) Possell, M., and Hewitt, C. N.: Isoprene emissions from plants
+        !       are mediated by atmospheric co2 concentrations, Global Change
+        !       Biology, 17, 1595-1610, 2011.
+
+        ! !REVISION HISTORY:
+        !  (1 ) Implemented in the standard code by A. Tai (Jun 2012).
+        !  See https://github.com/geoschem/hemco for complete history
+        !EOP
+        !------------------------------------------------------------------------------
+        !BOC
+
+        !-----------------------
+        ! Compute GAMMA_CO2
+        !-----------------------
+
+        !----------------------------------------------------------
+        ! Choose between two alternative CO2 inhibition schemes
+        !----------------------------------------------------------
+
+        IF ( co2_opt .eq. 0 ) THEN
+
+            ! Use empirical relationship of Possell & Hewitt (2011):
+            ! Empirical relationship of Possell & Hewitt (2011) based on nine
+            ! experimental studies including Wilkinson et al. (2009). This is
+            ! especially recommended for sub-ambient CO2 concentrations:
+
+            GAMMA_CO2 = 8.9406_rk / ( 1.0_rk + 8.9406_rk * 0.0024_rk * co2_set )
```

Comment:
I think you mentioned that this is supposed to always be < 1. But [this is](https://www.wolframalpha.com/input?i=solve+8.9406+%2F+%28+1.0+%2B+8.9406+*+0.0024+*+x%29+%3D+1) only less than one if `co2_set` is greater than ~ 370 ppmv.

Reply:
If the formula is correct, may want to cap it to max value 1.

Reply:
Looks like this is needed for opt 1 as well.

```
   0.0000000000000000        8.9405999999999999        1.5375360000000002        1.0000000000000000
   80.000000000000000        3.2911049831789443        1.4382684736450873        1.0000000000000000
   160.00000000000000        2.0167417126952185        1.3245077479049618        1.0000000000000000
   240.00000000000000        1.4538067798656267        1.2095607748916908        1.0000000000000000
   320.00000000000000        1.1365582505235445        1.0982758349857775        1.0000000000000000
   400.00000000000000       0.93296696141156998       0.99307552083334438        1.0000000000000000
   480.00000000000000       0.79123356468606532       0.89496569473289334        1.0000000000000000
   560.00000000000000       0.68688427339097324       0.80409070271143057        1.0000000000000000
   640.00000000000000       0.60685160343597777       0.73940373546497307        1.0000000000000000
   720.00000000000000       0.54352277166241547       0.68960115740854899        1.0000000000000000
   800.00000000000000       0.49216248066712942       0.63710179325366523        1.0000000000000000
```
:point_up: First col CO2, other cols gammas for opts 0, 1, 2

Reply:
@zmoon My statement may have been a bit vague, and maybe indeed it should only be < 1 for CO2 concentrations higher than a base value, i.e., 400 ppmv.  In HEMCO MEGANv3 implementation, it is not forced  to be a max value of 1.

To elucidate this further, in MEGANv3:
```
CO2temp = CO2
      Ci = 0.7* CO2

      WHERE (CO2temp.eq.400.0)
      GAMCO2 = 1.0
      ELSEWHERE
      GAMCO2 = ISmax- ((ISmax*Ci**CO2h) /(Cstar**CO2h+Ci**CO2h))
      ENDWHERE
```

You can see GAMCO2 is only set to 1.0 when CO2=400, but can be either < 1 or > 1 at different CO2 concentrations.

Thus, I think we should leave it as is in the code now.

Reply:
Ok. The fancier method seems fine but the Wilkinson factors seem pretty unrealistic at the very low CO2. Maybe a small warning somewhere?

Reply:
Maybe somewhere in the readme, in the table or below.

Reply:
OK, I put it in the updated [README.](https://github.com/noaa-oar-arl/canopy-app/pull/73/commits/a42c9a858e847bb589436a723fa14c8dd4c5f2f2)

Reply:
Oops, my bad. Wilkinson (opt 1) is the "fancier method". I think that should be the one recommended for sub-ambient based on those results printed out.

Reply:
OK, I updated the README based on the results.